simplesource-command-kafka/src/main/java/io/simplesource/kafka/internal/streams/EventSourcedTopology.java

```diff
@@ -1,14 +1,14 @@
 package io.simplesource.kafka.internal.streams;
 
 import io.simplesource.api.Aggregator;
-import io.simplesource.api.InitialValue;
```

Comment:
Maybe we should move this file into the new `topologies` folder.

Also, it feels like we've partially broken this file up. Is this part complete?

There's still a lot going on - I prefer a class / object to be either a worker or a delegator, and preferably not both.

As an alternative to breaking up a big class into smaller ones to make it easier to test, an alternative is to separate the state out completely (into a value class - it's all final), say `TopologyContext`, and then make everything else static. That's my preference, but I realise it's a question of style, what's done here is also valid.

But then I'd like to see this class broken up completely so it only delegates.

Reply:
At the end of this refactoring, `EventSourcedTopology` will be a delegator to all other sub-topologies(`CommandProcessingTopology` is one of them).

This `EventSourcedTopology` has to be public class as it will be accessible by `EventSourcedStreamsApp`. All other sub-topology classes will be package-private